File: 2654_minOperations/sol.c
int gcd(int a, int b) {
    while (b != 0) {
        int temp = b;
        b = a % b;
        a = temp;
    }
    return a;
}

int minOperations(int* nums, int numsSize) {
    int countOnes = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == 1) countOnes++;
    }
    
    if (countOnes > 0) {
        return numsSize - countOnes;
    }
    
    int minLen = numsSize + 1;
    
    for (int i = 0; i < numsSize; i++) {
        int g = nums[i];
        for (int j = i; j < numsSize; j++) {
            g = gcd(g, nums[j]);
            if (g == 1) {
                int len = j - i + 1;
                if (len < minLen) minLen = len;
                break;
            }
        }
    }
    
    if (minLen == numsSize + 1) {
        return -1;
    }
    
    return minLen - 1 + numsSize - 1;
}

File: 2654_minOperations/sol.py
from math import gcd
from typing import List

class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        
        count_ones = sum(1 for num in nums if num == 1)
        
        if count_ones > 0:
            return n - count_ones
        
        min_len = n + 1
        
        for i in range(n):
            g = nums[i]
            for j in range(i, n):
                g = gcd(g, nums[j])
                if g == 1:
                    min_len = min(min_len, j - i + 1)
                    break
        
        if min_len == n + 1:
            return -1
        
        return min_len - 1 + n - 1